src/server/game/AI/SmartScripts/SmartAI.cpp

```diff
@@ -453,45 +453,15 @@ void SmartAI::MoveInLineOfSight(Unit* who)
 
     GetScript()->OnMoveInLineOfSight(who);
 
-    if (me->HasReactState(REACT_PASSIVE) || AssistPlayerInCombat(who))
+    if (AssistPlayerInCombat(who))
         return;
 
-    if (!CanAIAttack(who))
-        return;
-
-    if (!me->CanStartAttack(who, false))
```

Comment:
I think this should not be removed, it's not part of CreatureAI::MoveInLineOfSight(who); and then it's useless to modify CanAIAttack method when its not called there